tap_salesforce/client.py

```diff
@@ -187,9 +190,29 @@ def get_records(
 
         query = f"{select_stm}{from_stm}{where_stm}{order_by_stm}{limit_stm}"
 
-        yield from self._paginate(
-            "GET", f"/services/data/{self._API_VERSION}/queryAll/", params={"q": query}
-        )
+        try:
+            yield from self._paginate(
+                "GET", f"/services/data/{self._API_VERSION}/queryAll/", params={"q": query}
+            )
+        except requests.exceptions.HTTPError as e:
+            nth = (end_date - start_date).total_seconds() / shrink_window_factor
```

Comment:
this will also swallow every other api error that is not a timeout issue. There should be a field in the error payload that should allow us to only perform this iff it is in fact a timeout.